packages/components/src/button/index.js

```diff
@@ -44,19 +58,56 @@ export function Button( props, ref ) {
 		'is-busy': isBusy,
 		'is-link': isLink,
 		'is-destructive': isDestructive,
+		'has-text': !! icon && !! children,
+		// Ideally should be has-icon but this is named this way for BC
```

Comment:
I thought classes didn't qualify?

https://github.com/WordPress/gutenberg/blob/master/docs/designers-developers/developers/backward-compatibility/README.md#class-names-and-dom-updates

Reply:
Yes, we can do it, just trying to keep the changes small for now. Ideally, we also deprecate the component.

Reply:
I looked at this and there are a few components that rely on this for styling. I'm going to leave it for a follow-up PR to ensure we test these properly and ease review of the current PR.